server.test.js

```diff
@@ -1,45 +1,126 @@
+    };
 
-  // test the PUT here
+    beforeEach(done => {
+        // write a beforeEach hook that will populate your test DB with data
+        // each time this hook runs, you should save a document to your db
+        // by saving the document you'll be able to use it in each of your `it` blocks
+        new Game({
+            title: 'Jackal',
+            genre: 'Top Scrolling Shooter',
+            releaseDate: 'June 27th, 1986'
+        }).save((err, savedGame) => {
+            if (err) {
+                console.log(err);
+                return done();
+            }
+            gameId = savedGame.id;
+            done();
+        });
+    });
+    afterEach(done => {
+        // simply remove the collections from your DB.
+        Game.remove()
+            .then(() => done())
+            .catch(err => done(err));
+    });
 
-  // --- Stretch Problem ---
-  // Test the DELETE here
+    // test the POST here
+    describe(`[POST] /api/game/create`, () => {
+        it('should create a new game', done => {
+            chai
+                .request(server)
+                .post('/api/game/create')
+                .send(newGame)
+                .then(res => {
+                    expect(res.status).to.equal(201);
+                    expect(res.body.game.title).to.equal('Final Fantasy');
+                    done();
+                })
+                .catch(err => done(err));
+        });
+    });
+    // test the GET here
+    describe(`[GET] /api/game/get`, () => {
+        it('should return all the games', done => {
+            chai
+                .request(server)
+                .get('/api/game/get')
+                .then(res => {
+                    expect(res.status).to.equal(200);
+                    expect(res.body).to.be.an('array');
+                    expect(res.body.length).to.equal(1);
+                    expect(res.body[0].title).to.equal('Jackal');
+                    done();
+                })
+                .catch(err => done(err));
+        });
+    });
+    // test the PUT here
+    describe(`[PUT] /api/game/update`, () => {
+        it('should update the game with the given id', done => {
+            chai
+                .request(server)
+                .put(`/api/game/update/${gameId}`)
+                .send(update)
+                .then(res => {
+                    expect(res.status).to.equal(200);
+                    expect(res.body.title).to.equal('Donkey Kong');
+                    done();
+                })
+                .catch(err => done(err));
+        });
+    });
+    // --- Stretch Problem ---
+    // Test the DELETE here
+    describe(`[DELETE] /api/game/destroy/:id`, () => {
+        it('should delete the game with the given id', done => {
+            chai
+                .request(server)
+                .delete(`/api/game/destroy/${gameId}`)
+                .then(res => {
+                    expect(res.status).to.equal(200);
```

Comment:
You'll probably also want to check to make sure that the game is actually no longer in the database.

Reply:
I saw that after the tests run the database is dropped/deleted from mongo so I didn't think you test for that as normally you wouldn't delete the whole database. Documentation I wanted to go back and add to after completing the tests, but ran short on time.